README.md

```diff
@@ -86,6 +86,51 @@ const SomeComponent = () => {
 };
 ```
 
+## Additional features
+
+### Caching
+
+There is the option to invalidate the cache, meaning next time the hook is called it will fetch the data again.
+
+By setting a timeTillCacheInvalidate time in ms, as follows:
+
+**In `SomeHighLevelComponent.tsx`**
+
+```tsx
+import React from 'react';
+import { Provider } from 'react-redux';
+import { store } from './redux/store';
+
+const SomeComponent = () => {
+  <Provider store={store}>
+    <ReactUseFetchWithReduxProvider timeTillCacheInvalidate={1800000}>
+      <App />
+    </ReactUseFetchWithReduxProvider>
+  </Provider>;
+};
+```
+
+Will result in the cache invalidating after 30 minutes.
+There is also the option to set the cache invalidation time per hook, with an optional third parameter like:
+
+**In `useThing.ts`**
+
+```typescript
+import { useFetchWithRedux } from 'react-use-fetch-with-redux';
+import { getThingStart } from './actions/ThingActions'; // getThingStart is an action creator.
+import { getThingSelector } from './selectors/ThingSelector'; // getThingSelector is a selector.
+
+const useThing = () =>
+  useFetchWithRedux(getThingStart, getThingSelector, {
+    timeTillCacheInvalidate: 1800000,
+  });
+
+export { useThing };
+```
+
+This will any value set by the provider set.
+You can also not set any value at the provider level, and handle all invalidation times in the hooks, but you will still need the provider, just with no value.
```

Comment:
Is this sentence true? I thought we were going to have it so that we could use the hook without any provider?

Reply:
It's only needed for caching. You can use it without but to remember when it was last used requires context, the way it is currently working.

You could maybe set this in state for each hook if we needed to do it like this, but then it wouldn't work across different instances of the same hook and cache keys.

Reply:
No that's cool. We just need to have a way of using this hook without the cache behaviour. Since you've confirmed that this is the case, we're good :)